main.cpp

```diff
@@ -103,24 +121,37 @@ struct HeapA
     }
 };
 
-#include <iostream>
-#include <cmath>
-#include <functional>
-#include <memory>
-
 
 template <typename NumericType>
 struct Numeric
 {
+    
     using Type = Temporary<NumericType>;
 
     Numeric(Type num) : numberPtr(std::make_unique<Type>(num)){}
 
+    Numeric(NumericType num) : numberPtr(std::make_unique<Type>(num)){}
+
+    template<typename OtherType>
+    Numeric& operator= (const OtherType& o)
+    {
+        *numberPtr = static_cast<NumericType>(o); 
+        return *this; 
+    }
+
```

Comment:
Remove these functions.
You have added another converting constructor `Numeric(NumericType)` and a converting assignment operator, which are not needed, and just add clutter.
- [ ] revise